relay-config/src/config.rs

```diff
@@ -222,6 +222,8 @@ trait ConfigObject: DeserializeOwned + Serialize {
 pub struct OverridableConfig {
     /// The operation mode of this relay.
     pub mode: Option<String>,
+    /// The deployment of this relay.
+    pub deployment: Option<String>,
```

Comment:
This is heavily conflicting with multiple terminologies, especially in the context of Kubernetes. The docs do not describe what this is supposed to contain but even then I would heavily refrain from using the word deployment for this.

Reply:
I am up for suggestions, for me `environment` was good, @jjbayer was leaning more towards deployment.